### PR TITLE
Tw refactor.2

### PR DESCRIPTION
**Pull request by siosonel**
## Description

Note: `serverconfig.features.usextw: true` is required to test the demonstration usage in matrix plot. There will be a `---`--- using xtw ---` message in the browser console when that feature flag is on.

This update
- puts tw categorical subclasses into one file, to minimize the number of files
- started to create numeric classes, organized in the same way as categorical subclasses
- prototyped composite classes as an option to extend only one base class, instead of multiple ones for a group of related TW types, such as by mode (discrete TW class versus a continuous TW class)
- extends composite xtw classes with setCellProps() methods in matrix plot, to test migrated term types + q.types/modes, unmigrated tws are not affected

Tested with http://localhost:3000/testrun.html?name=*.unit, http://localhost:3000/testrun.html?name=*.integration with `serverconfig.features.usextw: true`, and by looking at various matrix plot example URLs. NOTE: Command line tests, `cd client; npm run test:[unit | integration]` will run without `xtw`, since the feature.usextw is not set to true in `server/test/serverconfig.json` .

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by airenzp — APPROVED** [review]
Looks good to me after discussing the changes. It would be nice to remove the old folder. Thanks for the hard work and I am fine if you merge when you think is ready.

File: client/tw/categorical.ts [hunk 1]
```diff
@@ -55,10 +58,68 @@ export class CategoricalBase {
...
+			   specialized fill() function directly.
+
+			2. The fill() function must fill-in any expected missing values,
+			   validate the tw.q shape at runtime, and throw on any error or mismatched expectation.
+			   Runtime validation is required because the input raw tw can come from anywhere,
+			   like term.bins.default, which is a runtime variable that is not possible to statically check.
+
+			3. The filled-in tw, when returned, must be **coerced** to the full NumTW* type, 
+			   in order to match the function signature's return type.
+		*/
```
Comment (gavrielm) [inline client/tw/categorical.ts:109]: Duplicated comment

File: client/tw/categorical.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		else if (q.type != 'values') throw `expecting tw.q.type='values', got ${tw.q.type}`
+
+		// GDC or other dataset may allow missing term.values
+		if (!term.values) term.values = {}
+		const numVals = Object.keys(tw.term.values).length
+		// GDC or other dataset may allow empty term.values
+		//if (!numVals) throw `empty term.values`
+		if (q.mode == 'binary') {
+			if (Object.keys(tw.term.values).length != 2) throw 'term.values must have exactly two keys'
+			// TODO:
```
Comment (gavrielm) [inline client/tw/categorical.ts:158]: `tw.term.values` should not be checked for `q.mode=binary`. `tw.q.customset` should be checked instead. Can address in a subsequent PR.

**Review by congyu-lu — APPROVED** [review]
works good for matrix

**Review by gavrielm — APPROVED** [review]

**Pull request merged**